core/storage/exploration/gae_models_test.py

```diff
@@ -316,3 +332,119 @@ def test_get_at_least_editable(self):
             exploration_models.ExpSummaryModel
             .get_at_least_editable('nonexistent_id'))
         self.assertEqual(0, len(exploration_summary_models))
+
+
+class StateIdMappingModelUnitTest(test_utils.GenericTestBase):
+    """Tests for the StateIdMappingModel."""
+
+    def test_create_successfully_with_new_id(self):
+        exploration_models.StateIdMappingModel.create(
+            exp_id='id_0',
+            exp_version=0,
+            state_names_to_ids={},
+            largest_state_id_used=0)
+
+        observed_model = (
+            exploration_models.StateIdMappingModel.
+            get_state_id_mapping_model('id_0', 0))
+
+        self.assertEqual(observed_model.state_names_to_ids, {})
+        self.assertEqual(observed_model.largest_state_id_used, 0)
+
+    def test_create_successfully_if_id_exists_and_overwrite(self):
```

Comment:
Hi @sophiewu6, this isn't quite right -- you're just repeating the same check twice here. Instead, a test of this form should:

- create a model (setup)
- retrieve the model and check that some data is equal to an expected value (pre-condition)
- create a model to overwrite the existing one (action)
- retrieve the model and check that the data is equal to the new value (post-condition)

The pre-condition and post-condition checks should be doing the same thing, and the pre-condition check should come before the action. See https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests

Reply:
Oh right, I misunderstood the overwriting part. Sorry about that. Please take a look now.